Yumemi-Weather/Yumemi-Weather/Controller/MainViewController.swift

```diff
@@ -34,22 +38,43 @@ final class MainViewController: UIViewController {
     }
     
     @objc func viewWillEnterForeground(_ notification: Notification) {
-        let result = weatherModel.fetchWeather()
-        
-        self.handleWeatherChange(result: result)
+        weatherView.activityIndicatorViewStartAnimating()
+        weatherModel.fetchWeather { result in
+            DispatchQueue.main.async {
+                self.weatherView.activityIndicatorViewStopAnimating()
+                self.handleWeatherChange(result: result)
+            }
+        }
     }
     
     func handleWeatherChange(result: Result<Response, Error>) {
         switch result {
-        case .success(let responce):
-            weatherView.set(response: responce)
+        case .success(let response):
+            weatherView.set(response: response)
         case .failure(let error):
             handleErrorOccurred(error: error)
         }
     }
     
     func handleErrorOccurred(error: Error) {
-        var message: String
+        let message = errorMessageFrom(error: error)
+        let alertController: UIAlertController = UIAlertController(
+            title:"Error",
+            message: message,
+            preferredStyle: .alert
+        )
+        let defaultAction: UIAlertAction = UIAlertAction(
+            title: "Close",
+            style: .default,
+            handler: nil
+        )
+        
+        alertController.addAction(defaultAction)
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    func errorMessageFrom(error: Error) -> String {
```

Comment:
```suggestion
    func errorMessage(from error: Error) -> String {
```
[NITS] 上の様にすると使う側で下記の様に使えます
```
	let message = errorMessage(from: error)
```
また、メソッド名としては、errorMessageは名詞なので動詞+名詞の形にするとよいかと